src/oneD/StFlow.cpp

```diff
@@ -147,6 +147,13 @@ void StFlow::setTransport(Transport& trans)
     m_trans = &trans;
     m_do_multicomponent = (m_trans->transportType() == "Multi");
 
+    if (!m_do_multicomponent && m_do_soret) {
+        throw CanteraError("setTransport",
+                           "Thermal diffusion (the Soret effect) is "
+                           "enabled, and requires using a multicomponent "
+                           "transport model.");
+    }
```

Comment:
So if we are doing this check here, can we get rid of the one in `StFlow::enableSoret`? That would make it possible to toggle these flags in either order, and all that matters is that they are consistent once you call `solve`, which seems like an improvement to me.

Reply:
I don't think we can. The check here is only for the case where Soret is enabled and you're trying to set the transport back to mixture averaged. I think if we want to be able to set the flags in either order (i.e., set `m_do_soret = True` when `m_do_multicomponent == False` or set `m_do_multicomponent = False` when `m_do_soret == True`), then we should automatically enable/disable multicomponent transport. For instance, what if the user enables Soret diffusion with the mixture averaged model and then tries to compute the diffusion coefficients? Is that affected by the Soret option?

I think it would be better to make sure the transport is always in a consistent state, either by forcing the user to manage it, or by doing it in the code. Since there are effectively only three options for consistency, it shouldn't be hard to manage automatically, if we'd prefer that. I guess I have a mild preference for making the user do it, given that I chose to write an exception rather than just manage it (also, the exception was easier to write 😄)

Reply:
But I should also note that I have little experience with managing this part of the code, so I'll defer to your expertise if you think we should remove the other check.

Reply:
I agree that the user should always be expected to set a valid combination of flags, with no magic that changes either the `soret_enabled` flag or the `m_do_multicomponent` flag. But in the case where the user is switching from (no soret & no multicomponent) to (soret & multicomponent) or vice versa, I think we can let them set those flags in either order, rather than forcing them to do it in a particular order. And I think that this test here is sufficient to achieve that.

I don't think there is any issue with the user doing other things besides calling the `solve` method -- all the methods for calculating transport properties are independent of the settings of the `Sim1D` object. For example, the Soret diffusion coefficients can be calculated iff a multicomponent transport object is loaded, whether or not the attached `Sim1D` object is set to use mixture averaged or multicomponent properties.

Reply:
Just to make sure I'm understanding everything: I don't think just this test is sufficient. If the state is (no soret & no multicomponent), and the test is removed from `enableSoret`, the user can enable Soret and run solve, there's no check that multicomponent is enabled (unless you're suggesting I add a check in the solve method?) On the other hand, if the state is (soret & multicomponent), and the user tries to set mixture-averaged, and the test I've added here is not in place, then the state becomes (soret & no multicomponent) and the user can go and solve with no exceptions raised (until the simulation fails). Moreover, in that case (provided the test in the enableSoret function is still in place) the user cannot even turn off Soret without first turning multicomponent back on and turning off Soret first, so its a very confusing state of affairs (I realize you're not suggesting I remove this new check).

I guess I don't see a way (without having another check somewhere else) to allow the flags to be specified in either order and remain in a consistent state. Am I missing something?

Also, what happens in `Sim1D` when Soret diffusion coefficients are enabled but the underlying Phase is mixture-averaged? How does it get diffusion coefficients, and if they're the same as the "normal" mixture averaged coefficients, why does a simulation that runs fine for a regular mixture-averaged case fail when Soret is incorrectly enabled?